Assets/FPS/Scripts/Gameplay/Nautica/BotKillerAgent.cs

```diff
@@ -69,187 +70,203 @@ public override void OnEpisodeBegin()
 			enemies = trainingLevelManager.enemies;
 			pickups = trainingLevelManager.pickups;
 
-            trackedHealth = 1f;
+			trackedHealth = 1f;
 			trackedEnemyHealth.Clear();
 			foreach (var e in enemies) trackedEnemyHealth.Add(1f);
+		}
+
+		private void DebugLogReward(float reward, string rewardtype)
+		{
+			string rewardOutput = "<b>Reward:</b> ";
+			if (reward < 0) rewardOutput = "<color=red>";
+			if (reward > 0) rewardOutput = "<color=green>";
+			rewardOutput += rewardtype + " <b>" + reward.ToString() + "</b></color>, STEP: " + StepCount.ToString() + " Cumulative: " + GetCumulativeReward().ToString();
+			Debug.unityLogger.Log(LOGTAG, rewardOutput);
+		}
+
+		public override void CollectObservations(VectorSensor sensor)
+		{
+			ObserveHealth(sensor);
+			ObserveAmmo(sensor);
+			ObserveEnemies(sensor);
+			ObservePickups(sensor);
+		}
+
+		private void ObserveHealth(VectorSensor sensor)
+		{
+			var healthComponent = GetComponent<Unity.FPS.Game.Health>();
+			float normalizedHealth = healthComponent.GetRatio();
+			sensor.AddObservation(normalizedHealth);
+			UpdateTrackedAgentHealthAndRewardAgent(normalizedHealth);
+		}
+
+		private void UpdateTrackedAgentHealthAndRewardAgent(float agentHealth)
+		{
+			if (agentHealth != trackedHealth)
```

Comment:
one way you can simplify this is to use a "guard clause"
since all the code is nested inside an if statement, you can use the opposite condition to exit out early.

so `if (agentHealth == trackedHealth) return`
then the rest of the code doesn't need to be nested, and there's less cognitive load
(cognitive load is a real thing -- when dealing with production projects it becomes more important)

Reply:
On Guard! 🤺